OpenKeychain/src/main/java/org/sufficientlysecure/keychain/ui/keyunlock/activities/CreateKeyWizardActivity.java

```diff
@@ -0,0 +1,301 @@
+        mCreateKeyWizardViewModel.updateWizardStateOnBack();
+    }
+
+    /**
+     * Updates the interface and the viewModel state when the user clicks on the next button.
+     *
+     * @param view
+     */
+    public void onNextClicked(View view) {
+        if (mCurrentVisibleFragment != null && mCurrentVisibleFragment.onNextClicked()) {
+            mCreateKeyWizardViewModel.updateWizardStateOnNext();
+            updateWizardState();
+        }
+    }
+
+    /**
+     * Callback for when the user confirms his unlock keyword (passphrase, pin, etc)
+     * This method is only called when there is a double keyword confirmation.
+     */
+    @Override
+    public void onNewUnlockKeywordConfirmed() {
+        mCreateKeyWizardViewModel.updateWizardStateOnNext();
+        updateWizardState();
+    }
+
+    /**
+     * User canceled the unlock method dialog, go to previous step.
+     */
+    @Override
+    public void onNewUnlockMethodCancel() {
+        mCreateKeyWizardViewModel.updateWizardStateOnBack();
+    }
+
+    /**
+     * Callback for when the user confirms his unlock pin
+     * This method is only called when there is an unlock request.
+     */
+    @Override
+    public void onUnlockRequest() {
+        Log.v(TAG, "onUnlockRequest");
+    }
+
+    @Override
+    public void onHideNavigationButtons(boolean hide) {
+        if (hide) {
+            mCreateKeyWizardActivityButtonContainer.setVisibility(View.INVISIBLE);
+            mCreateKeyWizardActivityButtonContainer.animate().setDuration(400);
+            mCreateKeyWizardActivityButtonContainer.animate().alpha(0);
+
+        } else {
+            mCreateKeyWizardActivityButtonContainer.setVisibility(View.VISIBLE);
+            mCreateKeyWizardActivityButtonContainer.animate().setDuration(400);
+            mCreateKeyWizardActivityButtonContainer.animate().alpha(1);
+        }
+    }
+
+    @Override
+    public void onAdvanceToNextWizardStep() {
+        mCreateKeyWizardViewModel.updateWizardStateOnNext();
+        updateWizardState();
+    }
+
+
+    /**
+     * Updates the model with the current selected unlock type.
+     *
+     * @param secretKeyType
+     */
+    @Override
+    public void setUnlockMethod(CanonicalizedSecretKey.SecretKeyType secretKeyType) {
+        mCreateKeyWizardViewModel.getWizardModel().setSecretKeyType(secretKeyType);
+    }
+
+    @Override
+    public void setPassphrase(Passphrase passphrase) {
+        mCreateKeyWizardViewModel.getWizardModel().setPassword(passphrase);
+    }
+
+    @Override
+    public void setUserName(CharSequence userName) {
+        mCreateKeyWizardViewModel.getWizardModel().setName(userName.toString());
+    }
+
+    @Override
+    public CharSequence getName() {
+        return mCreateKeyWizardViewModel.getWizardModel().getName();
+    }
+
+    @Override
+    public ArrayList<String> getAdditionalEmails() {
+        return mCreateKeyWizardViewModel.getWizardModel().getAdditionalEmails();
+    }
```

Comment:
this seems like a lot of unnecessary boilerplate. can't the data providers just activity.getModel().getName()?